huaweicloud/resource_huaweicloud_lb_certificate.go

```diff
@@ -85,66 +95,71 @@ func ResourceCertificateV2() *schema.Resource {
 	}
 }
 
-func resourceCertificateV2Create(d *schema.ResourceData, meta interface{}) error {
+func resourceCertificateV2Create(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
-	elbClient, err := config.ElbV2Client(GetRegion(d, config))
+	elbClient, err := config.LoadBalancerClient(GetRegion(d, config))
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud elb client: %s", err)
+		return fmtp.DiagErrorf("Error creating HuaweiCloud elb client: %s", err)
 	}
 
 	createOpts := certificates.CreateOpts{
-		Name:        d.Get("name").(string),
-		Description: d.Get("description").(string),
-		Type:        d.Get("type").(string),
-		Domain:      d.Get("domain").(string),
-		PrivateKey:  d.Get("private_key").(string),
-		Certificate: d.Get("certificate").(string),
+		Name:                d.Get("name").(string),
+		Description:         d.Get("description").(string),
+		Type:                d.Get("type").(string),
+		Domain:              d.Get("domain").(string),
+		PrivateKey:          d.Get("private_key").(string),
+		Certificate:         d.Get("certificate").(string),
+		EnterpriseProjectID: GetEnterpriseProjectID(d, config),
 	}
 
 	logp.Printf("[DEBUG] Create Options: %#v", createOpts)
 	c, err := certificates.Create(elbClient, createOpts).Extract()
 	if err != nil {
-		return fmtp.Errorf("Error creating Certificate: %s", err)
+		return fmtp.DiagErrorf("Error creating Certificate: %s", err)
 	}
 
 	// If all has been successful, set the ID on the resource
-	d.SetId(c.ID)
+	d.SetId(c.Id)
 
-	return resourceCertificateV2Read(d, meta)
+	return resourceCertificateV2Read(ctx, d, meta)
 }
 
-func resourceCertificateV2Read(d *schema.ResourceData, meta interface{}) error {
+func resourceCertificateV2Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
-	elbClient, err := config.ElbV2Client(GetRegion(d, config))
+	elbClient, err := config.LoadBalancerClient(GetRegion(d, config))
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud elb client: %s", err)
+		return fmtp.DiagErrorf("Error creating HuaweiCloud elb client: %s", err)
 	}
 
 	c, err := certificates.Get(elbClient, d.Id()).Extract()
 	if err != nil {
-		return CheckDeleted(d, err, "certificate")
+		return common.CheckDeletedDiag(d, err, "certificate")
```

Comment:
the exception information `certificate` is not clear, the log printed is: certificate: xxx